includes/sanitizers/class-amp-script-sanitizer.php

```diff
@@ -106,28 +101,8 @@ class AMP_Script_Sanitizer extends AMP_Base_Sanitizer {
 	public function init( $sanitizers ) {
 		parent::init( $sanitizers );
 
-		if (
-			array_key_exists( AMP_Style_Sanitizer::class, $sanitizers )
-			&&
-			$sanitizers[ AMP_Style_Sanitizer::class ] instanceof AMP_Style_Sanitizer
-		) {
-			$this->style_sanitizer = $sanitizers[ AMP_Style_Sanitizer::class ];
-		}
-
-		if (
-			array_key_exists( AMP_Img_Sanitizer::class, $sanitizers )
-			&&
-			$sanitizers[ AMP_Img_Sanitizer::class ] instanceof AMP_Img_Sanitizer
-		) {
-			$this->img_sanitizer = $sanitizers[ AMP_Img_Sanitizer::class ];
-		}
-
-		if (
-			array_key_exists( AMP_Form_Sanitizer::class, $sanitizers )
-			&&
-			$sanitizers[ AMP_Form_Sanitizer::class ] instanceof AMP_Form_Sanitizer
-		) {
-			$this->form_sanitizer = $sanitizers[ AMP_Form_Sanitizer::class ];
+		foreach ( $sanitizers as $sanitizer ) {
+			$this->sanitizers[ get_class( $sanitizer ) ] = $sanitizer;
 		}
```

Comment:
Is this necessary? Why not:

```suggestion
		$this->sanitizers = $sanitizers;
```

Reply:
I was thinking this would be a hardening, but now that I think about it, probably it does nothing.